### ISSUE TITLE
3.65 henkaku enso compatibility

### ISSUE THREAD

**Issue by nightwishfan1** [OperationNT414C/DSMotion#3]
Not sure if you're still active with developing homebrew for vita, but henkaku enso got ported to 3.65.  Would love to see your plugins updated so potentially more games may become more playable thanks to them.  Thank you.

**Comment by OperationNT414C** [comment]
Hello,
I am still active, from time to time (depends on my mood: playing or coding, that's the question :-) ).
I am aware of Enso Henkaku 3.65 and I already updated my PS Vita (and the module ShellSecBat by the way). But currently, I wish to keep my other system (the PS TV) on 3.60. And as DS4Vita has not been updated, I cannot test DSMotion on my PS Vita (on DS4Vita, controller binding doesn't work anymore and I did not have any controller bound before I update my PS Vita).
However, only plugins which plays with internal function patch (and relies on module offsets) should be updated. This is not the case of DSMotion which only hooks exposed functions. So DSMotion "should" work without any update (I do not guaranty at 100%). Did you test it on 3.65?
Thank you for your report.

**Comment by nightwishfan1** [comment]
I'm gonna give them a try tonight after work if I remember to.  If not then tomorrow morning.  Been busy all today, and finally had enough time to actually update to 3.65 henkaku enso on my pstv.  I'll keep you posted.

**Comment by Localhorst86** [comment]
~~Tried it on my PSTV updated to 3.65 (had to wait for the usbmc.skprx to be 3.65 compatible) and a DS4 controller. I tested it on TID PCSF00349 (Flower) and unfortunatelly it doesn't detect motion controls :(~~

scratch that, it works. I just had to switch the order of plugin loading around (i.e. load nonpdrm before dsmotion)

**Comment by nightwishfan1** [comment]
Well I can confirm it still seems to work with Uncharted Golden Abyss, which I have physically, and skipped to the river chapter where the motion control is essential to get through.  So physical games don't seem like they should be an issue.

My one question I do have that I can't quite figure out is how do I order sequential title ids with the plugin?  I know *ALL is the easiest method, but I'm going on a title id by title id basis with regards to adding them to config.txt.  So I imagine the config would have to look something like:

*PCSA00029
ur0:tai/dsmotion.suprx
*next title id
ur0:tai/dsmotion.suprx

Correct me if I'm wrong about that, cause I honestly don't know, and am assuming stacking title ids under the *PCSA00029 one for Uncharted wouldn't be appropriate unless you say otherwise.

**Comment by OperationNT414C** [comment]
I am glad it works fine on 3.65 because I don't have any mean to test it (as I said before, I don't wish to update my PS TV to 3.65 for the moment).
About controlling the titles where the module is started, you can simply add for each additional title:

*KERNEL
ux0:tai/dsmotion.skprx
*TITLEID00
ur0:tai/dsmotion.suprx
*TITLEID02
ur0:tai/dsmotion.suprx
*TITLEID01
ur0:tai/dsmotion.suprx

There doesn't need to be some specific order in title IDs. However, I don't know if there is an hardcoded limit of supported "title IDs/plugins" association count in Henkaku.

**Comment by OperationNT414C** [comment]
Works fine on 3.65 without any required change. Thank you again for the tests!

**Comment by dem1980** [comment]
Hello,
i have 3.065 H.encore and my dsmotion is broken with ds3 and ds4.
Here is my config.txt in ur0, could you confirme me the order ? Thx

*KERNEL
*main

*KERNEL
ur0:tai/nonpdrm.skprx
ur0:tai/dsmotion.skprx
ur0:tai/usbmc.skprx
*ALL
ur0:tai/dsmotion.suprx

**Comment by nightwishfan1** [comment]
Got your config a bit wrong there.  Kernel doesn't need to be commented twice, nor does it need *main after it.  Your mention of *kernel with the skprx plugins is fine.  Note I'm saying skprx plugins since they are kernel plugins.  The suprx plugin gets commented further down where it notes the following lines:

this is for modifying the version string in settings widget
ux0:app/MLCL00001/henkaku.suprx

Under the second line stating ux0:app or ur0:app (whichever you're using) is where you comment your *ALL with the suprx plugin under it like the following:

this is for modifying the version string in settings widget
ux0:app/MLCL00001/henkaku.suprx
*ALL
ur0:app/dsmotion.suprx

So to sum up....You only need the first mention of *KERNEL for commenting your skprx plugins under.  You don't need it commented twice, or essentially nothing you got on your system will work, cause you edited the config wrong.  The suprx plugin goes a fair bit below that with where I noted it needing to be above.  Note that the lines I'm mentioning with the comment that they are for "this is for modifying the version string in settings widget" have a hashtag before them.  I just haven't used them here, because they cause undesired text edits here.

**Comment by OperationNT414C** [comment]
Thank you nightwishfan1 for the reply.

Also, don't forget that there is an issue where the controller must be connected AFTER the plugin initialization:
https://github.com/OperationNT414C/DSMotion/issues/4
This issue is systhematic with H-encore and it can be reproduced with Enso when the controller is powered on early while PS TV is booting.
The easy by-pass is to power off/on the controller once H-encore has been started up.

By the way, you can also take the version "DSMotionKernelLogs.zip" available on this issue to get traces of controller connection.

I will try to find a clean way to fix this problem if it's possible.

**Comment by nightwishfan1** [comment]
I would, but I'm personally on 3.65 enso upgraded via The Flows 3.65 updater tool he released a while back, and don't have the issue that the user on 3.68 has since enso loads henkaku and it's config.txt with plugins on boot from what I can tell.  So I merely helped him out to the best of my knoweledge with how to sort it out since I don't use 3.68 firmware.  To me it's a non issue with the controller at all, but for those on 3.68 it probably would be, since as you said 3.68 henkaku via H-encore requires install each time from a powered off state, and plugins won't load until re-installed, and rebooted so it catches the controllers responses when it pairs.

Edit:  Since I don't meet the H-encore criteria, and the user who had the problem on 3.68 is most likely the one you're needing the test from then I'm providing his reddit profile https://www.reddit.com/user/jackolantern_ perhaps a private message to him there will expedite the test process.

If I could be more help I would, but again I'm not upgraded via H-encore.  Just The Flows 3.65 updater tool, that he released some time ago.

**Comment by dem1980** [comment]
Hi,
I'm also in 3.65 Enso on my PSTV.
I was in 3.60 Enso and DSmotion works great with uncharted (nonpdrm + usbmc + dsmotion) for example but i have to upgrade to 3.65 because one of my games needs 3.61.
And now DSmtion doesn't worked.
My new config.txt (nonpdrm and ubsmc are working) : 
*KERNEL
ur0:tai/nonpdrm.skprx
ur0:tai/usbmc.skprx
ur0:tai/dsmotion.skprx
*ALL
ur0:tai/dsmotion.suprx

I have put files from DSMotionKernelLogs.zip into my ur0:tai/ directory but i haven't the file "ux0:data/dsmotion_k_log.txt" so i guess that the plugin is not loading correctly...
I have also tested to power off/power on the DS3 and DS4 after boot without success.

**Comment by dem1980** [comment]
It is ok for me now.
I have erased the ur0:\tai and ux0:\tai from internal memory (ps tv) and the ux0:tai from usb device.
I have put dsmotion with logs and a clean config.txt (like in my previous post) and now i have log and the motion works !
I think that the ux0 from internal memory could affects the ux0 from usb device before the plugin usbmc is fully loaded...

**Comment by OperationNT414C** [comment]
About "tai/config.txt" file, it works like this:
- First: Sony official memory card
- Second: Internal memory, partition "imc0:"
- Last: Internal memory, partition "ur0:"
I think some plugins which mount a specific drive as "ux0:" (gamesd, usbmc, ...) try to reload the configuration file from the mounted drive but I won't rely on this too much.
